core/shared/src/test/scala/scalaz/zio/internal/StackBoolSpec.scala

```diff
@@ -1,78 +1,61 @@
 package scalaz.zio.internal
 
-import org.specs2.Specification
+import org.scalacheck.{ Arbitrary, Gen, Prop }
+import org.specs2.{ ScalaCheck, Specification }
 
-class StackBoolSpec extends Specification {
+class StackBoolSpec extends Specification with ScalaCheck {
   def is =
     "StackBoolSpec".title ^ s2"""
         Size tracking                 $e0
-        From/to list small identity   $e1
-        From/to list large identity   $e2
-        Small push/pop example        $e3
-        Large push/pop example        $e4
-        Peek/pop identity             $e5
+        From/to list identity         $e1
+        Push/pop example              $e2
+        Peek/pop identity             $e3
     """
 
-  def e0 = {
-    val list = List.fill(100)(true)
+  import Arbitrary._
 
-    StackBool(list: _*).size must_== list.length
-  }
+  private val generator: Gen[List[Boolean]] = boolListGen(0, 400)
```

Comment:
I might be wrong, but maybe 200 is big enough - there are 3 entries at this point.

Reply:
I've adjusted the max size according to your recommendations.